Riot/Modules/Application/AppCoordinator.swift

```diff
@@ -298,16 +298,39 @@ fileprivate class AppNavigator: AppNavigatorProtocol {
         return SideMenuPresenter(sideMenuCoordinator: sideMenuCoordinator)
     }()
     
+    private var appNavigationVC: UINavigationController {
+        guard
+            let splitVC = appCoordinator.splitViewCoordinator?.toPresentable() as? UISplitViewController,
+            // Picking out the first view controller currently works only on iPhones, not iPads
+            let navigationVC = splitVC.viewControllers.first as? UINavigationController
+        else {
+            MXLog.error("[AppNavigator] Missing root split view controller")
+            return UINavigationController()
+        }
+        return navigationVC
+    }
+    
     // MARK: - Setup
     
     init(appCoordinator: AppCoordinator) {
         self.appCoordinator = appCoordinator
-        self.alert = AppAlertPresenter(legacyAppDelegate: appCoordinator.legacyAppDelegate)
     }
     
     // MARK: - Public
     
     func navigate(to destination: AppNavigatorDestination) {
         self.appCoordinator.navigate(to: destination)
     }
+    
+    func addLoadingActivity() -> Activity {
+        let presenter = ActivityIndicatorToastPresenter(
+            text: VectorL10n.roomParticipantsSecurityLoading,
```

Comment:
For now we only use a single message, in the future we can customize this method with more parameters / viewState